backend/src/controllers/userController.ts

```diff
@@ -0,0 +1,82 @@
+import { PrismaClient } from "@prisma/client";
+import { Request, Response } from "express";
+import { generateNonce } from "siwe";
+import { errorResponse } from "../utils/utils";
+
+const prisma = new PrismaClient();
+
+/**
+ *
+ * @param req
+ * @param res
+ * @returns
+ */
+const createUser = async (req: Request, res: Response) => {
+  const { address } = req.params;
+  const nonce = generateNonce();
+
+  try {
+    if (address) {
+      await prisma.user.create({
+        data: {
+          address,
+          nonce,
+        },
+      });
+
+      await prisma.$disconnect();
+
+      res.status(200).json({
+        success: true,
+        nonce,
+      });
+    } else {
+      throw Error(`address empty, please user address`);
+    }
+  } catch (error) {
+    await prisma.$disconnect();
+
+    return errorResponse(500, error, res);
+  }
+};
+
+/**
+ *
+ * @param req
+ * @param res
+ * @returns
+ */
+const getNonce = async (req: Request, res: Response) => {
+  const { address } = req.params;
+
+  try {
+    if (!address)
+      return errorResponse(401, "Get nonce error: Missing user address", res);
+
+    const user = await prisma.user.findUnique({
+      where: {
+        address,
+      },
+    });
+
+    await prisma.$disconnect();
+
+    if (!user)
+      return res.status(200).json({
+        success: true,
+        nonce: false,
+      });
```

Comment:
I think you can generate a nonce for a user that not exit.

Reply:
In that case there's no need to make an API request, it can be done in the front-end using the library. If you want users that don't exist to be able to interact with the application then that's okay but then anyone with the `siwe` library can generate valid data and there's no way of ensuring that it's valid for _our_ platform.

Reply:
No, I'm just thinking how a user who does not have an account can receive a nonce to sign his message.